src/main/kotlin/com/autonomousapps/tasks/AdviceTask.kt

```diff
@@ -132,58 +137,88 @@ abstract class AdviceTask : DefaultTask() {
 
     val computedAdvice = advisor.compute(filterSpecBuilder())
     val advices = computedAdvice.getAdvices()
-    val advicePrinter = AdvicePrinter(computedAdvice)
-
-    var didGiveAdvice = false
-    val consoleReport = StringBuilder()
+    val consoleReport = ConsoleReport.build(computedAdvice)
 
-    if (!computedAdvice.filterRemove) {
-      advicePrinter.getRemoveAdvice()?.let {
-        consoleReport.add("Unused dependencies which should be removed:\n$it\n")
-        didGiveAdvice = true
-      }
-    }
+    adviceFile.writeText(advices.toJson())
+    advicePrettyFile.writeText(advices.toPrettyString())
+    adviceConsoleFile.writeText(consoleReport.toJson())
+    adviceConsolePrettyFile.writeText(consoleReport.toPrettyString())
 
-    if (!computedAdvice.filterAdd) {
-      advicePrinter.getAddAdvice()?.let {
-        consoleReport.add("Transitively used dependencies that should " +
-          "be declared directly as indicated:\n$it\n")
-        didGiveAdvice = true
-      }
-    }
+    chatter.chat("See machine-readable report at ${adviceFile.path}")
```

Comment:
These specifically should have only been done in the `AdvicePrinterTask`. I changed that.
The files are always output in the `AdviceTask`, we  print the location of the files on screen if we are chatty.
In the printer task, the `if` gates the message to be displayed. Let me know how you like this.

Reply:
It looks better. Some of the output seems redundant, but I don't really care right now (I think we can make a readability pass at some point in the future).

I am a bit confused by `advice-console.txt` and `advice-console-pretty.json`. Those files have similar names, but their contents are radically different. One looks basically like a json-prettified version of a `List<Advice>` (with names, so more like a Map I guess), while the other is a literal record of the console output.

In an attempt to move this forward, here's a concrete proposal. Let me know what you think.
1. Rename `advice-console-pretty.json` to `advice-simplified.json`
2. Print just the `Dependency` objects, rather than the full `Advice` object. As you've already noted, the extra info contained by `Advice` becomes redundant when you present it as a map like this.